test/powershell/engine/COM/COM.Basic.Tests.ps1

```diff
@@ -43,6 +48,47 @@ try {
         }
     }
 
+    Describe 'GetMember/SetMember/InvokeMember binders should have more restricted rule for COM object' -Tags "CI" {
+        BeforeAll {
+            $null = New-Item -Path $TESTDRIVE/bar -ItemType Directory -Force
+
+            $shell = New-Object -ComObject "Shell.Application"
+            $folder = $shell.Namespace("$TESTDRIVE")
+            $item = $folder.Items().Item(0)
+
+            ## Create a PSObject that has an instance member 'Name' and a script method 'Windows'
+            $str = Add-Member -InputObject "abc" -MemberType NoteProperty -Name Name -Value "Hello" -PassThru
+            $str = Add-Member -InputObject $str -MemberType ScriptMethod -Name Windows -Value { "Windows" } -PassThru
+        }
+
+        It "GetMember binder should differentiate PSObject that wraps COM object from other PSObjects" {
+            ## GetMember on the member name 'Name'.
+            ## '$_' here is a PSObject that wraps a COM object
+            $item | ForEach-Object { $_.Name } > $null
```

Comment:
I think it's useful to test with a single dynamic site, something like this:

```PowerShell
$t1 = ($item, "?")
$t2 = ($str, "Hello")
for ($pair in ($t1, $t2, $t2, $t1, $t1, $t2)) {
    $pair[0].Name | Should Be $pair[1]
}
```

It's not much different from what you have, but this way you know it's the site's cache that influences your test and not the process wide cache, plus you can test all the rule orderings explicitly.